### ISSUE TITLE
pylint 3.2.0 causes false-positive E1133 (not-an-iterable) error on many of more-itertools functions return values

### ISSUE THREAD

**Issue by ZivRonen** [pylint-dev/pylint#9626]
### Bug description

```python
"""Show pylint not-an-iterable issue with more-itertools. Pylint: 3.2.0, more-itertools: 1.20.0"""
import more_itertools

for val in more_itertools.chunked([1, 2, 3], n=1):
    print(val)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:4:11: E1133: Non-iterable value more_itertools.chunked([1, 2, 3], n=1) is used in an iterating context (not-an-iterable)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)

### Pylint version

```shell
pylint 3.2.0
astroid 3.2.0
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
```

### Additional dependencies

```python
more-itertools==1.20.0
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Caused in pylint-dev/astroid#2375, which we may need to revert.

**Comment by jacobtylerwalls** [comment]
Planning to get patch releases of astroid and pylint out in the next few days. This should have been guarded under an experimental feature flag.

**Comment by ZivRonen** [comment]
Ok, thank you very much 👍
Is it grouped under another issue? Should I close this one?

**Comment by jacobtylerwalls** [comment]
Let's leave it open so that it's discoverable until we get the patches out.

**Comment by ZivRonen** [comment]
Fixed by astroid version 3.2.1

**Comment by Pierre-Sassoulas** [comment]
If you have pylint 3.2.0 you need to upgrade astroid (``pip install astroid -U``) or wait for pylint 3.2.1 to be released.
